### PR TITLE
feat(jellyfish-api-core): add verifyMessage and signMessageWithPrivKey RPCs

### PR DESCRIPTION
**Pull request by shohamc1**
#### What this PR does / why we need it:

Adds verifyMessage and signMessageWithPrivKey RPCs and tests.

#### Which issue(s) does this PR fixes?:

Fixes part of https://github.com/JellyfishSDK/jellyfish/issues/48

### CODE REVIEW

File: packages/jellyfish-api-core/__tests__/category/misc/signMessageWithPrivKey.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      payload: {
+        code: -5,
+        message: 'Invalid private key',
+        method: 'signmessagewithprivkey'
+      }
+    })
+  })
+
+  it('should sign with private key', async () => {
+    const promise = client.misc.signMessageWithPrivKey('cRiRQ9cHmy5evDqNDdEV8f6zfbK6epi9Fpz4CRZsmLEmkwy54dWz', 'test')
```
Comment (shancht) [inline packages/jellyfish-api-core/__tests__/category/misc/signMessageWithPrivKey.test.ts]: What are your thoughts on the following lines instead for `line 32`?

```
    const privkey = await client.wallet.dumpPrivKey(await client.wallet.getNewAddress())
    const promise = client.misc.signMessageWithPrivKey(privkey, 'test')
```

**Review by shancht — CHANGES_REQUESTED** [review]

File: packages/jellyfish-api-core/__tests__/category/misc/signMessageWithPrivKey.test.ts (hunk 1 above)
Comment (shohamc1) [reply packages/jellyfish-api-core/__tests__/category/misc/signMessageWithPrivKey.test.ts]: Seems like a much better way to do it, fixed in https://github.com/JellyfishSDK/jellyfish/pull/1905/commits/3ff9fe86f89fd768fb2cc6b6334c4aa1c1e1d02e

**Review by shancht — DISMISSED** [review]
Looks good to me!

File: packages/jellyfish-api-core/__tests__/category/misc/signMessageWithPrivKey.test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { RpcApiError } from '@defichain/jellyfish-api-core'
+
+describe('Verify message', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
```
Comment (marktanrj) [inline packages/jellyfish-api-core/__tests__/category/misc/signMessageWithPrivKey.test.ts]: nitpick: `waitForReady()` is deprecated and can be removed, `start()` will automatically wait for ready

image

File: packages/jellyfish-api-core/__tests__/category/misc/verifyMessage.test.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { RpcApiError } from '@defichain/jellyfish-api-core'
+
+describe('Verify message', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+  const message = 'test'
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
```
Comment (marktanrj) [inline packages/jellyfish-api-core/__tests__/category/misc/verifyMessage.test.ts]: Same for this

**Review by marktanrj — DISMISSED** [review]
Looks good, left some comments

**Review by marktanrj — APPROVED** [review]
LGTM

**Review by shancht — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**